### ISSUE TITLE
Invalid "headers" settings for Bombardier

### ISSUE THREAD

**Issue by alvpickmans** [dotnet/crank#364]
When using bombardier to test a POST endpoint with a json body, `--header` argument value is wrapped on curly braces but according to `bombardier --help` it should just be `--header "key: value"`.

crank-agent log:
```
[...]
[12:02:16.655] Starting -> Running (load:20)
[12:02:16.655] args: -c 125 -w 15 -d 60 -n 50 --insecure -l --http1 --header {
[12:02:16.655]   content-Type: application/json
[12:02:16.655] } http://localhost:5000/postendpoint -f ./payload.json -m POST
[...]
```` 

Error thrown by crank 
```
Errors:
    Post "http://localhost:5000/executions": net/http: invalid header field name "{content-Type" - 6145230
```

Bombardier help
```
bombardier --help
usage: C:\bombardier\bombardier.exe [<flags>] <url>

Fast cross-platform HTTP benchmarking tool

Flags:
  [...]
  -f, --body-file=""          File to use as request body
  -s, --stream                Specify whether to stream body using chunked
                              transfer encoding or to serve it from memory
      --cert=""               Path to the client's TLS Certificate
      --key=""                Path to the client's TLS Certificate Private Key
  -k, --insecure              Controls whether a client verifies the server's
                              certificate chain and host name
  -a, --disableKeepAlives     Disable HTTP keep-alive. For fasthttp use -H
                              'Connection: close'
  -H, --header="K: V" ...     HTTP headers to use(can be repeated) 👈
  [...]
```

**Comment by corrieriluca** [comment]
> `--header` argument value is wrapped on curly braces

How does look like your YAML configuration file? It would help to see how you filled the `customHeaders` variable.

> according to `bombardier --help` it should just be `--header "key: value"`

When using Bombardier with Crank, the Crank Agent is not calling directly the Bombardier command, it uses a [client/wrapper made in C#](https://github.com/dotnet/crank/blob/main/src/Microsoft.Crank.Jobs.Bombardier/Program.cs) which has its own set of CLI options (close to Bombardier options but not identical) and has the responsibility of:
- Downloading the appropriate Bombardier binary according to the OS/Architecture on which Crank is running
- Setting a bunch of event sources for collecting metrics
- Launching Bombardier with the appropriate options

**Comment by alvpickmans** [comment]
Something that I noticed from the Bombardier wrapper is that on the Bombardier wrapper, the [MeasureFirstRequest()](https://github.com/dotnet/crank/blob/4d0d673bb79c7e3cf2f9cddd5159c43808598de4/src/Microsoft.Crank.Jobs.Bombardier/Program.cs#L318) method defaults to using `HttpMethod.Get` which in my case will fail as that endpoint is POST only.

So even if the headers are properly applied, it would not pass beyond that line as the first request is likely to fail.

**Comment by sebastienros** [comment]
It should respect the verb, good find. Do you think that will fix your issue?

**Comment by alvpickmans** [comment]
I would assume the first request might still fail because as far as I can see, headers are not being sent as part of the HttpMessageRequest?

After that, if headers are properly sent to bombardier wrapper even if the crank-agent logs invalid arguments (ie `--header {content-Type: application/json}` instead of `--header "content-Type: application/json"`), I think it should work. I'd still expect logs to honour the arguments specs on bombardier though.

I'll be glad to help if I get a chance to familiarise with the code base.

**Comment by corrieriluca** [comment]
@alvpickmans Recently I tested setting the `customHeaders` variable in the Bombardier job, because, similarly to you, I needed to set the `Content-Type` header to `application/json`. It turns out that you must pass the headers this way (just an array of strings):

```yaml
customHeaders: [ 'Content-Type: application/json' ]
```

And not `customHeaders: [ 'Content-Type': 'application/json' ]` as you did. This is a huge difference in YAML syntax as it converts to some sort of dictionary, hence the invalid argument you get (`--header { content-Type: application/json }`).

In the logs of Crank Agent, Bombardier receives the right arguments:
```
bombardier  -d 1s -c 16 -t 2s --insecure -l --http1 --header "Content-Type: application/json"
```
